### PR TITLE
intermodal: update to 0.1.12.

### PR DESCRIPTION
**Pull request by jcgruenhage**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### tl;dr: computers are hard
So, the `imdl` package was a bit barebones so far, but I'm not quite sure what the correct way to fix this is. Intermodal does have really nice tooling for automatically generating additional artifacts (completions, man pages, other docs, etc) and does try to make packaging `imdl` easy, but sadly, this additional tooling is not really built with cross compilation in mind (afaict), and it requires access to the git history of the repository (for changelog generation). It would be possible to work around these limitations here, but it probably makes sense to instead think about how to improve this, approach upstream about it (they say "First off, thank you very much! If I can do anything to make packaging Intermodal easier, please don't hesistate to open an issue." in the repo upstream and provide a lot of useful info in the readme already), improve the situation upstream and then use the improved tooling in a package here.

I have still extended the package a tiny bit to ship shell completions, because this is the part that bugged me the most, and doing so was fairly straight forward using our vtargetrun helper. I've cargo-culted those bits from our rustup package, I hope this is the right way to go?

cc @casey (upstream) and @ThatNerdyPikachu (void contributor for this package)

### CODE REVIEW

File: srcpkgs/intermodal/template [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
 maintainer="Pika <pika@lasagna.dev>"
 license="CC0-1.0"
 homepage="https://github.com/casey/intermodal"
 distfiles="https://github.com/casey/intermodal/archive/v${version}.tar.gz"
-checksum=8bb3e4f549e5c4446543babd741fec0cd7c42da4d49eca3e98c5f7611ad59618
+checksum=cd62894e519dc5aa0284a5f48aab86e1a45c3bc96b8a5481741adb6960d4751a
+
+post_build() {
+	IMDL="target/${RUST_TARGET}/release/imdl"
+	ln -sf "${IMDL}" imdl
```
Comment (ahesford) [inline srcpkgs/intermodal/template:17]: The link is superfluous...

File: srcpkgs/intermodal/template [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+
+	for shell in zsh bash fish; do
+		vtargetrun ./imdl completions ${shell} >imdl.${shell}
```
Comment (ahesford) [inline srcpkgs/intermodal/template:20]: ...because you can just run"$IMDL" right here.

Comment (jcgruenhage) [reply srcpkgs/intermodal/template:20]: cargo-culted from our rustup template, but yeah, seems a bit redundant to have both the variable *and* the link

**Comment by ahesford** [comment]
Cloning the git repo is generally undesirable with xbps-src. The best fix from our perspective would be for upstream to include pre-generated docs in the release tarballs. This requires a bit of release engineering but shouldn't be too terrible and can be scripted for future releases.

In general, I think maintaining a separate changelog is good practice for several reasons:

1. It doesn't require repository history to review.
2. It provides an opportunity to add context to some more sophisticated changes.
3. Not all Git commits are worth inclusion in history. Changelogs should provide a good summary of substantive changes, not every minor tweak. (Does it really matter to others that you fixed a typo in one commit that you introduced in the preceding commit?)


For packages that require doc generation, we have sometimes used a hostmakedepends trick to get those docs in cross-compiled packages. For native packages, the docs are built as expected; for cross builds, the package itself is added to hostmakedepends. The docs can then be copied from the native package on the host to the $DESTDIR. Yes, this is a hack, but it works.

**Comment by jcgruenhage** [comment]
> For packages that require doc generation, we have sometimes used a hostmakedepends trick to get those docs in cross-compiled packages. For native packages, the docs are built as expected; for cross builds, the package itself is added to hostmakedepends. The docs can then be copied from the native package on the host to the $DESTDIR. Yes, this is a hack, but it works.

Are you able to link a package that does work like this? Just so that I can do some more cargo-culting, to get everything except the changelog generated using the upstream tooling.

**Comment by Johnnynator** [comment]
I don't know what might be a good example. What ahesford is suggesting is essentially to build the docs on your machine. cp it to `srcpkgs/intermodal/files/smth` and just do `vdoc "${FILESDIR}/smt"` in the template.
Ofcourse replace `smth` with whatever file(s) you have.

The downside of this, is that the generated docs have to be manually updated if the package gets updated.

**Comment by ahesford** [comment]
I think @ericonr did this for documentation in some package that I can't remember. A few other templates do this because the  build process requires native executables to generate other build products. `protobuf`, `gnubg` and `postfix` all pull in the native package with
```sh
if [ "$CROSS_BUILD" ]; then
    hostmakedepends+=" ${pkgname}"
fi
```
which would be sufficient to give you native documentation that you could copy in `post_install`.

**Comment by jcgruenhage** [comment]
I don't know which of the two alternatives I find worse tbh. Wasting time on building the native package a dozen times is shit, but having to manually rebuild docs is terrible as well. Having them distributed by upstream isn't much better either IMO, as they somewhat are compilation artifacts, and we should be able to build them ourselves...

As it's "just" compute time that I'd be wasting with the `hostmakedepends` variant and not human time, I'd go with that route for now (or rather, will do when I have time, busy with other stuff for now.

Afaict, that gives us everything except a changelog, so that'd be the highest priority to fix upstream IMO. Other opinions on this here?

**Comment by ahesford** [comment]
The only "waste" here is because our build infrastructure doesn't properly handle dependency resolution. Eventually™ this problem will go away and I don't think we should be too aggressive in codifying the current limitations of the build infrastructure into templates. I recommend the `hostmakedepends` trick because I think it's less of a hassle than maintaining our own copies of files that will ultimately become outdated.

**Comment by jcgruenhage** [comment]
CI is still busy on this one, but I've locally done a cross build from scratch and the content of the resulting package looks good to me (diffed the content with the natively build package, just the binary and props.plist differ). The native build is tested, man pages and completions are there as expected and look good, am fairly happy with it now.

The only thing that I'm still kinda missing is a changelog now, but as discussed before, this is something we can't really fix downstream. Anything else we'd want to do here, or is this basically ready to be merged now?
